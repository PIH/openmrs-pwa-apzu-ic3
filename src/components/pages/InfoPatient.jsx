import React from "react";
import { connect } from "react-redux";

import { Grid, Row, Col } from 'react-bootstrap';

class InfoPatient extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <form>
                <Grid>
                    <Row>
                        <Col>
                            <label>ID: </label>
                            <input type="text" name="patientId" defaultValue={this.props.patient.id}/>
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <label>UUID: </label>
                            <input type="text" name="uuid" defaultValue={this.props.patient.uuid}/>
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <label>Identifier: </label>
                            <input type="text" name="uuid" defaultValue={this.props.patient.identifier}/>
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <label>First Name: </label>
                            <input type="text" name="patientId" defaultValue={this.props.patient.firstName}/>
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <label>Last Name: </label>
                            <input type="text" name="patientId" defaultValue={this.props.patient.lastName}/>
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <label>Gender: </label>
                            <input type="text" name="patientId" defaultValue={this.props.patient.gender}/>
                        </Col>
                    </Row>
                    <Row>
                        <Col>
                            <label>Age: </label>
                            <input type="text" name="patientId" defaultValue={this.props.patient.age}/>
                        </Col>
                    </Row>
                </Grid>
            </form>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        patient: state.selected.patient
    };
};

export default connect(mapStateToProps)(InfoPatient);