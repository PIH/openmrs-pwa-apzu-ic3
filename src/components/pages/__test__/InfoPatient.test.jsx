import React from 'react'
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer'
import InfoPatient from '../InfoPatient'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'

let props, mountedComponent, store;

const mockStore = configureMockStore();
const infoPatient = () => {
    if (!mountedComponent) {
        mountedComponent = mount(
            <Provider store={store}>
                <InfoPatient {...props} />
            </Provider>);
    }
    return mountedComponent;
};

describe('Component: InfoPatient', () => {
    beforeEach(() => {
        props = {};
        store = mockStore(
            {
                selected: {
                    patient: {
                        id: 1,
                        lastName: "Wilkin",
                        firstName: "Dessin",
                        gender: "M",
                        age: 25,
                    }
                }
            });
        mountedComponent = undefined;
    });

    it('renders properly', () => {
        expect(toJson(infoPatient())).toMatchSnapshot();
        expect(infoPatient().find(InfoPatient).length).toBe(1);
    });

    it('renders a form element', () => {
        expect(infoPatient().find('form').length).toBe(1);
    });

});

