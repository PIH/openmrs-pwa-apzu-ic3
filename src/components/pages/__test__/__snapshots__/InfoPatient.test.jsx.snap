// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: InfoPatient renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(InfoPatient)>
    <InfoPatient
      dispatch={[Function]}
      patient={
        Object {
          "age": 25,
          "firstName": "Dessin",
          "gender": "M",
          "id": 1,
          "lastName": "Wilkin",
        }
      }
    >
      <form>
        <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}
        >
          <div
            className="container"
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                >
                  <div
                    className=""
                  >
                    <label>
                      ID: 
                    </label>
                    <input
                      defaultValue={1}
                      name="patientId"
                      type="text"
                    />
                  </div>
                </Col>
              </div>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                >
                  <div
                    className=""
                  >
                    <label>
                      UUID: 
                    </label>
                    <input
                      name="uuid"
                      type="text"
                    />
                  </div>
                </Col>
              </div>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                >
                  <div
                    className=""
                  >
                    <label>
                      Identifier: 
                    </label>
                    <input
                      name="uuid"
                      type="text"
                    />
                  </div>
                </Col>
              </div>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                >
                  <div
                    className=""
                  >
                    <label>
                      First Name: 
                    </label>
                    <input
                      defaultValue="Dessin"
                      name="patientId"
                      type="text"
                    />
                  </div>
                </Col>
              </div>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                >
                  <div
                    className=""
                  >
                    <label>
                      Last Name: 
                    </label>
                    <input
                      defaultValue="Wilkin"
                      name="patientId"
                      type="text"
                    />
                  </div>
                </Col>
              </div>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                >
                  <div
                    className=""
                  >
                    <label>
                      Gender: 
                    </label>
                    <input
                      defaultValue="M"
                      name="patientId"
                      type="text"
                    />
                  </div>
                </Col>
              </div>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <div
                className="row"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                >
                  <div
                    className=""
                  >
                    <label>
                      Age: 
                    </label>
                    <input
                      defaultValue={25}
                      name="patientId"
                      type="text"
                    />
                  </div>
                </Col>
              </div>
            </Row>
          </div>
        </Grid>
      </form>
    </InfoPatient>
  </Connect(InfoPatient)>
</Provider>
`;
